shell/platform/fuchsia/flutter/tests/integration/text-input/text-input-view/lib/text_input_view.dart

```diff
@@ -8,14 +8,98 @@
 import 'package:fuchsia_services/services.dart';
+import 'package:zircon/zircon.dart';
+
+// Corresponds to the USB HID values provided in fidl.fuchsia.input
+// https://fuchsia.dev/reference/fidl/fuchsia.input
+final Map<int, String> hidToKey = {
+  458977: 'LEFT_SHIFT', // Keyboard Left Shift
+  458792: 'ENTER', // Keyboard Enter (Return)
+  458782: 'KEY_1', // Keyboard 1 and !
+  458763: 'H', // Keyboard h and H
+  458760: 'E', // Keyboard e and E
+  458767: 'L', // Keyboard l and L
+  458770: 'O', // Keyboard o and O
+  458778: 'W', // Keyboard w and W
+  458773: 'R', // Keyboard r and R
+  458759: 'D', // Keyboard d and D
+};
 
 int main() {
-  // TODO(https://fxbug.dev/107917): Port https://cs.opensource.google/fuchsia/fuchsia/+/main:src/ui/tests/integration_input_tests/text-input/text-input-flutter/lib/text-input-flutter.dart
-  // to dart:ui.
-  print('text-input-view: starting');
+  print('Launching text-input-view');
+  TestApp app = TestApp();
+  app.run();
+}
+
+class TestApp {
+  static const _yellow = Color.fromARGB(255, 255, 255, 0);
+  Color _backgroundColor = _yellow;
+
+  final _responseListener = test_text.KeyboardInputListenerProxy();
+
+  void run() {
+    // Connect to keyboard input response listener
+    Incoming.fromSvcPath().connectToService(_responseListener);
+    // Set up window callbacks
+    window.onPlatformMessage = (String name, ByteData data, PlatformMessageResponseCallback callback) {
+      this.platformMessage(name, data);
+    };
+    window.onMetricsChanged = () {
+      window.scheduleFrame();
+    };
+    window.onBeginFrame = (Duration duration) {
+      this.beginFrame(duration);
+    };
+
+    window.scheduleFrame();
+  }
+
+  void beginFrame(Duration duration) {
+    // Convert physical screen size of device to values
+    final pixelRatio = window.devicePixelRatio;
+    final size = window.physicalSize / pixelRatio;
+    final physicalBounds = Offset.zero & size * pixelRatio;
+    final windowBounds = Offset.zero & size;
+    // Set up a Canvas that uses the screen size
+    final recorder = PictureRecorder();
+    final canvas = Canvas(recorder, physicalBounds);
+    canvas.scale(pixelRatio, pixelRatio);
+    // Draw something
+    final paint = Paint()..color = this._backgroundColor;
+    canvas.drawRect(windowBounds, paint);
+    // Build the scene
+    final picture = recorder.endRecording();
+    final sceneBuilder = SceneBuilder()
+      ..pushClipRect(physicalBounds)
+      ..addPicture(Offset.zero, picture)
+      ..pop();
+    window.render(sceneBuilder.build());
+  }
+
+  void platformMessage(String name, ByteData data) async {
+    final buffer = data.buffer;
+    var list = buffer.asUint8List(data.offsetInBytes, data.lengthInBytes);
+    var decoded = utf8.decode(list);
+    var decodedJson = json.decode(decoded);
+    print('received ${name} platform message: ${decodedJson}');
+
+    if (name == "flutter/keyevent" && decodedJson["type"] == "keydown") {
```

Comment:
Should we also have a test for the `flutter/textinput` channel? My impression was that we send messages along both `flutter/keyevent` and `flutter/textinput`.